packages/mobile/src/harmony-native/foundations/typography/Text.tsx

```diff
@@ -116,21 +28,33 @@ export const Text = (props: TextProps) => {
     color: colorProp = 'default',
     ...other
   } = props
-  const styles = useStyles()
   const theme = useTheme()
   // TODO: make heading a proper gradient
   const color =
     colorProp === 'heading'
       ? theme.color.secondary.secondary
       : theme.color.textIcon[colorProp]
 
-  const styleKey = `${variant}-${size}-${strength}`
+  const variantStyles = variantStylesMap[variant]
+  // const variantStyles = variantStylesMap.label
+
+  const textStyles = useMemo((): TextStyle => {
+    const t = theme.typography
+    return {
+      fontSize: t.size[variantStyles.fontSize[size]],
+      lineHeight: t.lineHeight[variantStyles.lineHeight[size]],
+      fontWeight: t.weight[variantStyles.fontWeight[strength]],
+      fontFamily: t.fontByWeight[variantStyles.fontWeight[strength]],
+      // @ts-ignore
+      ...(variantStyles.css ?? {})
```

Comment:
this is tough with the types.... any thoughts on what's best here? i think maybe you dont need to ts-ignore when its passed to css prop?